File: chapter02/selectionSort.py
# 选择排序

# 找出数组中最小的元素
def findSmallest (arr):
	smallest = arr[0]
	smallest_index = 0
	for i in range(1, len(arr)):
		if arr[i] < smallest:
			smallest = arr[i]
			smallest_index = i
	return smallest_index

# 对数组进行排序
def selectionSort(arr):
	newArr = []
	for i in range(len(arr)):
		smallest = findSmallest(arr)
		newArr.append(arr.pop(smallest))
	return newArr

print(selectionSort([5, 3, 6, 2, 10]))

File: chapter02/selectionSort.java
// 选择排序
public class selectionSort {
	public static void main(String[] args) {

	}

	public static int findSmallest(ArrayList arr) {
		smallest = arr[0];
		smallest_index = 0;
		for (int i = 1; i < arr.length; i++) {
			if (arr[i] < smallest) {
				smallest = arr[i];
				smallest_index = i;
			}
		}
		return smallest_index;
	}

	public static int[] selectionSort(ArrayList arr) {
		ArrayList newArr = new ArrayList();
		

	}
}